internal/kms/azure_vault.go

```diff
@@ -0,0 +1,225 @@
+	certs, key, err := azidentity.ParseCertificates([]byte(kms.clientCertificate), []byte{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse Azure client certificate: %w", err)
+	}
+	creds, err := azidentity.NewClientCertificateCredential(kms.tenantID, kms.clientID, certs, key, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Azure credentials: %w", err)
+	}
+
+	azClient, err := azsecrets.NewClient(kms.vaultURL, creds, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Azure client: %w", err)
+	}
+
+	return azClient, nil
+}
+
+func (kms *azureKeyVaultKMS) getSecrets() (map[string]interface{}, error) {
+	c, err := k8s.NewK8sClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to kubernetes to "+
+			"get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
+	}
+
+	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
+		kms.secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
+	}
+
+	config := make(map[string]interface{})
+	for k, v := range secret.Data {
+		switch k {
+		case azureClientCertificate:
+			config[k] = string(v)
+		default:
+			return nil, fmt.Errorf("unsupported option for KMS provider %q: %s", kmsTypeAzure, k)
+		}
+	}
+
+	return config, nil
+}
+
+// FetchDEK returns passphrase from Azure key vault.
+func (kms *azureKeyVaultKMS) FetchDEK(key string) (string, error) {
+	svc, err := kms.getService()
+	if err != nil {
+		return "", fmt.Errorf("failed to get KMS service: %w", err)
+	}
+
+	getSecretResponse, err := svc.GetSecret(context.TODO(), key, "", nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to get secret: %w", err)
+	}
+
+	return *getSecretResponse.Value, nil
+}
+
+// StoreDEK saves new passphrase to Azure key vault.
+func (kms *azureKeyVaultKMS) StoreDEK(key, value string) error {
+	svc, err := kms.getService()
+	if err != nil {
+		return fmt.Errorf("failed to get KMS service: %w", err)
+	}
+
+	setSecretParams := azsecrets.SetSecretParameters{
+		Value: &value,
+	}
+	_, err = svc.SetSecret(context.TODO(), key, setSecretParams, nil)
+	if err != nil {
+		return fmt.Errorf("failed to set seceret %w", err)
+	}
+
+	return nil
+}
+
+// RemoveDEK deletes passphrase from Azure key vault.
+func (kms *azureKeyVaultKMS) RemoveDEK(key string) error {
+	svc, err := kms.getService()
+	if err != nil {
+		return fmt.Errorf("failed to get KMS service: %w", err)
+	}
+
+	_, err = svc.DeleteSecret(context.TODO(), key, nil)
+	if err != nil {
+		responseError, ok := err.(*azcore.ResponseError)
+		if ok && responseError.StatusCode == 404 {
+			return PurgeSecret(svc, key)
+		}
+		return fmt.Errorf("failed to delete seceret %w", err)
+	}
+
+	return PurgeSecret(svc, key)
+}
+
+// PurgeSecret removes the secret permanently from the Azure key vault.
+func PurgeSecret(svc *azsecrets.Client, key string) error {
```

Comment:
this is an internal helper function, make sure not to export it and call it `purgeSecret`. Maybe include Azure in the name, as it is still part of the `kms` package.

Reply:
as purging is not needed, I have removed this now.